File: Frontend/trackr-administrador/src/app/shared/http/catalogo/localidad.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Localidad } from '@models/catalogo/localidad';

@Injectable()
export class LocalidadService{
    private dataUrl = 'localidad/';

    constructor(public http: HttpClient) {}

    consultarPorEstado(idEstado: number): Observable<Localidad[]> {
      return this.http.get<Localidad[]>(this.dataUrl + `consultarPorEstado/${idEstado}`);
    }
}

File: Backend/TrackrAPI/TrackrAPI/Services/Catalogo/LocalidadService.cs
using TrackrAPI.Models;
using TrackrAPI.Repositorys.Catalogo;

namespace TrackrAPI.Services.Catalogo
{
    public class LocalidadService
    {
        private ILocalidadRepository localidadRepository;

        public LocalidadService(
            ILocalidadRepository localidadRepository
        )
        {
            this.localidadRepository = localidadRepository;
        }

        public IEnumerable<Localidad> ConsultarPorEstado(int idEstado)
        {
            return localidadRepository.ConsultarPorEstado(idEstado);
        }
    }
}
